### DISCUSSION TITLE
Compile boringssl with CMake

### DISCUSSION THREAD

**Discussion by b-wille** [ngtcp2/ngtcp2#1378] [Q&A] [answered]
Hello,

Unfortunately it is not possible for me to compile the boringssl libs with CMake. I always get the following error message.

CMake Warning at crypto/CMakeLists.txt:31 (message):
  libngtcp2_crypto_quictls library is disabled due to lack of good quictls

CMake Warning at crypto/CMakeLists.txt:43 (message):
  libngtcp2_crypto_boringssl library is disabled due to lack of good
  BoringSSL

How can I compile the libs for boringssl and quictls with CMake?

I would be very happy to receive an answer.
Thanks

Greetings

**Comment by tatsuhiro-t** [comment, marked as answer]
Check https://github.com/ngtcp2/ngtcp2/blob/8d5ab7874f7b8846f6e398f0a76069e4e5766d70/.github/workflows/build.yml#L509-L510 that is how we build the lib with boringssl.

**Reply by b-wille** [reply]
Hello

many thanks for the quick response.

I can get it built slightly modified with the Alpine Package Builder (abuild), but unfortunately not with OPENWRT/Entware Package Builder. But the most important thing for me was that it was confirmed to me that boringssl/aws-lc can also be built with CMake, I will be able to solve the problems somehow.

I still have one question. Can you actually build boringssl/aws-lc and quictls together in one go?

I would be happy to receive an answer.
Thanks

Greetings

**Reply by tatsuhiro-t** [reply]
You can build with quictls and either boringssl or aws-lc.  You cannot build with both boringssl and aws-lc at the same time.

**Reply by b-wille** [reply]
Sorry, my mistake, I should have written this more precisely. That's exactly what I mean, quictls + (aws-lc or boringssl) + gnutls ....

Thank you.
